File: Lab7/DictionaryTest.java
// Connor Monson cmonson@ucsc.edu
// lab7 
// DictionaryTest.java

public class DictionaryTest {

   public static void main(String[] args) {
   
      Dictionary D = new Dictionary();
         System.out.println(D.isEmpty());
         D.insert("1","f");
         System.out.println(D.isEmpty());
         D.insert("5","h");
         System.out.println(D.size());
         D.insert("3","a");
         System.out.println("*********");
         System.out.println(D.size());
         System.out.println("*********");
      
         D.insert("8","c");
         D.insert("2","e");
         D.insert("9","3");
      
         //D.delete("1");
         System.out.println(D);
         System.out.println(D.size());
         D.delete("3");
         System.out.println(D.size());
         System.out.println(D);
         System.out.println(D.lookup("5"));
      
         //D.insert("1", "f"); //throws DuplicateKeyException
         //D.delete("10"); //throws KeyNotFoundException
         D.makeEmpty();
         System.out.println(D.size() + " " + D.isEmpty());
   }
}

File: Lab5/DictionaryTest.c
/* 
*  Connor Monson cmonson@ucsc.edu 1461818
*  this is lab5 essencially doing pa3 in c
*  fall 2016
*  DictionaryTest.c 
*  Tests Dictionary.h using Hash Tables
*/

#include<stdio.h> //include everything
#include<stdlib.h>
#include<string.h>
#include"Dictionary.h" //test Dictionary.h using Hash Tables
#define MAX_LEN 180 //set max length

int main(int argc, char* argv[]) {
   Dictionary T = newDictionary(); //initilize
   char* c;
   char* m;
   char* R[] = {"one","two","three","four","five","six","seven"}; //char arrays
   char* S[] = {"a","b","c","d","e","f","g"};
   int i; //initilize for loop

   for(i=0; i<7; i++) { //loop to insert()
      insert(T, R[i], S[i]);
   }
   
   printDictionary(stdout, T);

   for(i=0; i<7; i++){
      c = R[i];
      m = lookup(T, c);
      printf("key=\"%s\" %s\"%s\"\n", c, (m==NULL?"not found ":"value="), m);
   }

   delete(T, "one");
   printDictionary(stdout, T);

   printf("%s\n", (isEmpty(T)?"true":"false"));
   printf("%d\n", size(T));
   makeEmpty(T);
   printf("%s\n", (isEmpty(T)?"true":"false"));
   freeDictionary(&T);

   return(EXIT_SUCCESS);
}